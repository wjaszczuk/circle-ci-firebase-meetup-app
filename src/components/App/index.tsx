import * as React from 'react';
import logo from './logo.svg';
import './index.scss';

const App: React.FunctionComponent = () => (
  <div className="app">
    <header className="app__header">
      <img src={logo} className="app__logo" alt="logo" />
      <p>
        Edit <code>src/App.js</code> and save to reload.
      </p>
      <a
        className="app__link"
        href="https://reactjs.org"
        target="_blank"
        rel="noopener noreferrer"
      >
        Learn React
      </a>
    </header>
  </div>
);

export default App;
