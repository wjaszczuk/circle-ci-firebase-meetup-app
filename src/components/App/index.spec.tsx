import React from 'react';
import ReactDOM from 'react-dom';
import App from './index';
import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-enzyme';

describe('App', () => {
  let sut: ShallowWrapper;

  beforeEach(() => {
    sut = shallow(<App />);
  });

  it('will match snapshot', () => {
    expect(sut).toMatchSnapshot();
  });

  it('will contain image with classname app__logo', () => {
    expect(sut.find('img')).toHaveClassName('app__logo');
  });
});
