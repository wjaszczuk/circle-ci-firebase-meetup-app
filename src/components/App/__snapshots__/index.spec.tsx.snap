// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App will match snapshot 1`] = `
<div
  className="app"
>
  <header
    className="app__header"
  >
    <img
      alt="logo"
      className="app__logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="app__link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
